### ISSUE TITLE
Serverpod Flutter web app fails to build

### ISSUE THREAD

**Issue by CordMemescape** [serverpod/serverpod#2654]
**Describe the bug**
Building a Serverpod Flutter web app that uses `serverpod_auth_shared_flutter` causes the build to fail due to `serverpod_auth_shared_flutter`'s dependency on `image_cropper`. Building results in the following error:

```
flutter build web --no-tree-shake-icons

Target dart2js failed: ProcessException: Process exited abnormally with exit code 1:
../../../../.pub-cache/hosted/pub.dev/image_cropper_platform_interface-5.0.0/lib/src/models/cropped_file/html.dart:32:38:
Error: The method 'UnmodifiableUint8ListView' isn't defined for the class 'CroppedFile'.
 - 'CroppedFile' is from 'package:image_cropper_platform_interface/src/models/cropped_file/html.dart'
 ('../../../../.pub-cache/hosted/pub.dev/image_cropper_platform_interface-5.0.0/lib/src/models/cropped_file/html.dart').
      return Future<Uint8List>.value(UnmodifiableUint8ListView(_initBytes!));
                                     ^^^^^^^^^^^^^^^^^^^^^^^^^
Error: Compilation failed.
  Command: /Users/Development/flutter/bin/cache/dart-sdk/bin/dart --disable-dart-dev
  /Users/Development/flutter/bin/cache/dart-sdk/bin/snapshots/dart2js.dart.snapshot
  --platform-binaries=/Users/Development/flutter/bin/cache/flutter_web_sdk/kernel --invoker=flutter_tool
  -Ddart.vm.product=true -DFLUTTER_WEB_AUTO_DETECT=false -DFLUTTER_WEB_USE_SKIA=true
  -DFLUTTER_WEB_CANVASKIT_URL=https://www.gstatic.com/flutter-canvaskit/c9b9d5780da342eb3f0f5e439a7db06f7d112575/
  --native-null-assertions --no-source-maps -o
  /Users/Development/limetrack/limetrack/limetrack_flutter/.dart_tool/flutter_build/b5a10d1bc0aca699e8f9fe041f9225bf/app.dill --packages=.dart_tool/package_config.json --cfe-only
  /Users/Development/limetrack/limetrack/limetrack_flutter/.dart_tool/flutter_build/b5a10d1bc0aca699e8f9fe041f9225bf/main.dart
#0      RunResult.throwException (package:flutter_tools/src/base/process.dart:122:5)
#1      _DefaultProcessUtils.run (package:flutter_tools/src/base/process.dart:370:19)
<asynchronous suspension>
#2      Dart2JSTarget.build (package:flutter_tools/src/build_system/targets/web.dart:203:5)
<asynchronous suspension>
#3      _BuildInstance._invokeInternal (package:flutter_tools/src/build_system/build_system.dart:875:9)
<asynchronous suspension>
#4      Future.wait.<anonymous closure> (dart:async/future.dart:534:21)
<asynchronous suspension>
#5      _BuildInstance.invokeTarget (package:flutter_tools/src/build_system/build_system.dart:813:32)
<asynchronous suspension>
#6      Future.wait.<anonymous closure> (dart:async/future.dart:534:21)
<asynchronous suspension>
#7      _BuildInstance.invokeTarget (package:flutter_tools/src/build_system/build_system.dart:813:32)
<asynchronous suspension>
#8      FlutterBuildSystem.build (package:flutter_tools/src/build_system/build_system.dart:635:16)
<asynchronous suspension>
#9      WebBuilder.buildWeb (package:flutter_tools/src/web/compile.dart:92:34)
<asynchronous suspension>
#10     BuildWebCommand.runCommand (package:flutter_tools/src/commands/build_web.dart:230:5)
<asynchronous suspension>
#11     FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1408:27)
<asynchronous suspension>
#12     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:153:19)
<asynchronous suspension>
#13     CommandRunner.runCommand (package:args/command_runner.dart:212:13)
<asynchronous suspension>
#14     FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:420:9)
<asynchronous suspension>
#15     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:153:19)
<asynchronous suspension>
#16     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:364:5)
<asynchronous suspension>
#17     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:130:9)
<asynchronous suspension>
#18     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:153:19)
<asynchronous suspension>
#19     main (package:flutter_tools/executable.dart:93:3)
<asynchronous suspension>

Compiling lib/main.dart for the Web...                             13.8s
Error: Failed to compile application for the Web.
```

This is a known `image_cropper` issue and is resolved in `image_cropper 7.1.0`

https://github.com/hnvn/flutter_image_cropper/issues/505

**To Reproduce**
1. Create a new Serverpod app and include the `serverpod_auth_shared` dependency for the server, client and flutter app
2. Build the app for the web using `flutter build web --no-tree-shake-icons`
3. See error

**Expected behavior**
Expected the web app to build

**Serverpod versions**
 -  _CLI Version:_ 2.0.2
 - _Library version:_ 2.0.2

**Platform information**
macOS

**Additional context**
Looking through the Serverpod codebase, it looks like this might be resolved in the upcoming 2.1.0 release with the update to the image_cropper dependency although it might be worth checking whether `image_cropper: '>=7.0.0 <9.0.0'`  should be changed to `image_cropper: '>=7.1.0 <9.0.0'` to ensure that the fix is included.

In attempting to work around the issue, adding a dependency override of `image_cropper: ^8.0.2` results in a new error:

```
Target dart2js failed: ProcessException: Process exited abnormally with exit code 1:
../../../../.pub-cache/hosted/pub.dev/serverpod_auth_shared_flutter-2.0.2/lib/src/image_uploader.dart:72:9:
Error: No named parameter with the name 'cropStyle'.
        cropStyle: CropStyle.circle,
        ^^^^^^^^^
Error: Compilation failed.
```
This is due to breaking changes in image_cropper 7.0.0+. Taken from their release notes:

- WebUiSettings: is totally re-implemented, please see plugin document for more details
- cropImage(): move `cropStyle` and `aspectRatioPresets` into `AndroidUiSettings` and `IOUiSettings` for sake of clean and clarity.

Before 7.0.0:
```
    File croppedFile = await ImageCropper().cropImage(
      sourcePath: imageFile.path,
      cropStyle: CropStyle.circle,
      aspectRatioPresets: [
        CropAspectRatioPreset.square,
        CropAspectRatioPreset.original,
      ],
      uiSettings: [
        AndroidUiSettings(
          toolbarTitle: 'Cropper',
        ),
        IOSUiSettings(
          title: 'Cropper',
        ),
      ],
    );
 ```

From 7.0.0 onwards:
```
    File croppedFile = await ImageCropper().cropImage(
      sourcePath: imageFile.path,
      uiSettings: [
        AndroidUiSettings(
          toolbarTitle: 'Cropper',
          cropStyle: CropStyle.circle,
          aspectRatioPresets: [
            CropAspectRatioPreset.square,
            CropAspectRatioPreset.original,
          ],
        ),
        IOSUiSettings(
          title: 'Cropper',
          cropStyle: CropStyle.circle,
          aspectRatioPresets: [
            CropAspectRatioPreset.square,
            CropAspectRatioPreset.original,
          ],
        ),
      ],
    );
```

**Comment by Isakdl** [comment]
I believe this should be fixed already, we have a published beta version for the next release here: https://pub.dev/packages/serverpod/versions/2.1.0-beta.2

Could you see if you get the same problem with this version?

**Comment by Isakdl** [comment]
Just append the version to the activate command :) 

`dart pub global activate serverpod_cli 2.1.0-beta.2`

And make sure all your Serverpod dependencies are on this version!

**Comment by CordMemescape** [comment]
I can confirm that the issue is resolved by 2.1.0-beta.2.

:-D

As always, thanks @Isakdl.

**Comment by Isakdl** [comment]
Closing as fixed in 2.1.0
